### ISSUE TITLE
Streaming on HomeKit causes camera crash (HomeBridge logs included)

### ISSUE THREAD

**Issue by carson-autocicd** [hjdhjd/homebridge-unifi-protect#422]
**Describe The Problem:**
Hi there, if I just leave the video on the tile it seems to work fine. But as soon as I click the tile to go into full screen it "crashes" and the video goes offline on HomeKit (it has no effect on the unifi app and is fine there). If I open it again it will load the image immediately, then freeze and crash (unless I quickly exit at which point I have the old image). This is true for all my unifi cameras (i.e. doorbell, g4 pro, and g3 flex). 

As a side note, if someone rings the doorbell, HomeKit will notify the doorbell has been rung, but soon as you try to look at it, it dies immediately.

**To Reproduce:**
Access any camera directly either by clicking on the tile for any Unifi camera or by clicking on an alert (such as the doorbell ringing).

**Logs:**
Note: I removed the MAC Addresses

```
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4]: Connected to the Protect controller API (address: 10.10.10.166 mac:).
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4]: Connected to the UniFi realtime update events API.
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4]: Discovered camera: Driveway [UVC G4 Pro] (address: 10.10.10.92 mac:).
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4]: Discovered camera: Backyard [UVC G4 Pro] (address: 10.10.10.10 mac:).
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4]: Discovered camera: Side stairs [UVC G4 Pro] (address: 10.10.10.86 mac:).
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4]: Discovered camera: Garage [UVC G3 Flex] (address: 10.10.10.106 mac:).
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4]: Discovered camera: Doorbell [UVC G4 Doorbell] (address: 10.10.10.183 mac:).
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4]: Controller refresh interval set to 10 seconds.
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4] Doorbell [UVC G4 Doorbell]: Discovered doorbell message switch: LEAVE PACKAGE AT DOOR.
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4] Doorbell [UVC G4 Doorbell]: Discovered doorbell message switch: DO NOT DISTURB.
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4] Doorbell [UVC G4 Doorbell]: Discovered doorbell message switch: PRESS TO SUMMON DOG..
[6/12/2021, 12:53:28 PM] [UniFi Protect] UNVR [UNVR4] Doorbell [UVC G4 Doorbell]: Doorbell message set: PRESS TO SUMMON DOG..

node:99976) UnhandledPromiseRejectionWarning: Error: Command failed with ENOENT: ffmpeg -codecs
spawn ffmpeg ENOENT
at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
at onErrorNT (internal/child_process.js:469:16)
at processTicksAndRejections (internal/process/task_queues.js:84:21)
(node:99976) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag --unhandled-rejections=strict (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:99976) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Homebridge Configuration:**

```json
{
    "controllers": [
        {
            "address": "10.10.10.166",
            "password": "<hidden>",
            "username": "homebridge"
        }
    ],
    "platform": "UniFi Protect"
}.
```

**Screenshots:**

On the tile you get an image, but I don't know that it ever really updates:
![image](https://user-images.githubusercontent.com/61490462/121788420-03827380-cb82-11eb-8128-bba00ea89e29.png)

When you click the image:
![image](https://user-images.githubusercontent.com/61490462/121788513-adfa9680-cb82-11eb-80a2-d076a564ca54.png)

**Environment:**

* **Homebridge Version:**  1.3.4
* **Node Version:** 12.18.4
* **Homebridge-UniFi-Protect Plugin Version:** 4.4.0
* **UniFi Protect Version:** 1.18.1
* **UniFi Protect Controller Type:** UNVR
* **UniFi Protect Controller Firmware Version:** 1.4.9
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** <iOS 14.6 (iPhone & iPAD) & MacOS 11.4
* **Operating System and OS Version:** MacOS 10.3.16

**Note:**
My HomeBridge host is a Mac mini (late 2012) with 16gb of ram, 2.5ghz i5 and Intel HD Graphics 4000. It's running macOS High Sierra (nervous about updating it because of its age and what it may or may not do to HomeBridge).

**Comment by hjdhjd** [comment]
Thanks for the report. This looks pretty straightforward.

From the error, it seems that you don’t have a working ffmpeg installed. The plugin should install one as part of it’s dependencies on it’s own, depending on how you installed it. How did you install the plugin? Command line? Or did you use the Homebridge UI to do it?

**Comment by carson-autocicd** [comment]
Oh thanks for getting back so quickly! I installed it through the HomeBridge UI

**Comment by carson-autocicd** [comment]
I tried to find the installation logs so I could see what installed exactly, but there don't seem to be any, it only shows the command used:

`[6/12/2021, 12:40:32 PM] [Homebridge UI] Running Command: npm install -g homebridge-unifi-protect@latest`

**Comment by carson-autocicd** [comment]
I tried running npm install ffmpeg manually from the directory that homebridge lives in (I'm not sure if that's right) and got the following error:

```
npm WARN saveError ENOENT: no such file or directory, open '/Users/automation/.homebridge/package.json'
npm WARN enoent ENOENT: no such file or directory, open '/Users/automation/.homebridge/package.json'
npm WARN .homebridge No description
npm WARN .homebridge No repository field.
npm WARN .homebridge No README data
npm WARN .homebridge No license field.

+ ffmpeg@0.0.4
updated 1 package and audited 3 packages in 0.282s
found 0 vulnerabilities

```

**Comment by carson-autocicd** [comment]
Ok I've tried installing it instead in what I think is the correct NPM Modules directory that looks like this:
```
Automations-Mac-mini:node_modules automation$ ls
ffmpeg				homebridge-envisalink		npm
homebridge			homebridge-haiku-senseme
homebridge-config-ui-x		homebridge-unifi-protect
```

No errors, but it still fails in the same way. So then I tried installing it in the unifi-protect directory which now looks like this:

```
Automations-Mac-mini:homebridge-unifi-protect automation$ ls
LICENSE.md		config.schema.json	node_modules		package.json
README.md		dist			package-lock.json
```

Also no errors, but also no change in behaviour. I'm kind of just guessing now so I'll wait for your feedback :)

**Comment by hjdhjd** [comment]
Stop. :smile:

1. Uninstall the plugin.
2. Install the Homebridge UI.
3. Use the Homebridge UI to install the plugin.
4. Configure the plugin.
5. Restart Homebridge.

Your assumptions about how to install this plugin aren’t quite right, and you didn’t read the instructions in the documentation about how to specifically install it. The easiest thing to do…follow the above instructions. Good luck.

**Comment by carson-autocicd** [comment]
Sorry maybe I was not clear - I have Homebridge UI installed, and that's how I installed the plugin. I clicked plugins, searched for the Homebridge Unifi Protect plugin, enter the credentials for the UNVR, and restarted Homebridge. I do not know why FFMEG did not install and was not found. The command line that I shared above was not what I ran to install the plugin, it's what I grabbed from the Homebridge UI log.

That said I can uninstall and reinstall and see what happens.
